lib/private/converter.rb

```diff
@@ -51,6 +51,17 @@ def _convert(value, type, raise_coercion_error)
         else
           _convert(value, type.types[nil_idx == 0 ? 1 : 0], raise_coercion_error)
         end
+      elsif type.is_a?(T::Types::TypedHash)
+        unless value.respond_to?(:map)
+          raise T::Coerce::ShapeError.new(value, type)
+        end
+
+        value.map do |k, v|
+          [
+            _convert(k, type.keys, raise_coercion_error),
```

Comment:
I wonder why it's `.keys`. Is there multiple keys or multiple types for keys?

Reply:
Since the return value is a simple type (not array), I guess it's the type of the "keys" and "values" in this hash?